frontend/app/(tabs)/home.tsx

```diff
@@ -1,10 +1,78 @@
-import { Text, View } from 'react-native';
+import {
+  View,
+  ScrollView,
+  NativeSyntheticEvent,
+  NativeScrollEvent,
+} from 'react-native';
+import React, { useState, useCallback, useEffect, useRef } from 'react';
+import { CardData } from '../types/card';
+import { fetchCards, CARDS_PER_PAGE } from '../services/cardService';
+import { Card } from '../components/Card';
 
 const Index = () => {
+  const [cards, setCards] = useState<CardData[]>([]);
+  const [page, setPage] = useState(1);
+  const isLoadingRef = useRef(false);
+
+  const loadInitialCards = async () => {
+    try {
+      isLoadingRef.current = true;
+      const initialCards = await fetchCards(1, CARDS_PER_PAGE);
+      setCards(initialCards);
+    } catch (error) {
+      console.error('Error loading initial cards:', error);
+    } finally {
+      isLoadingRef.current = false;
+    }
+  };
+
+  const loadMoreCards = useCallback(async () => {
+    if (isLoadingRef.current) return;
+    try {
+      isLoadingRef.current = true;
+      const newCards = await fetchCards(page + 1, CARDS_PER_PAGE);
+      setCards((prev) => [...prev, ...newCards]);
+      setPage((prev) => prev + 1);
+    } catch (error) {
+      console.error('Error loading more cards:', error);
+    } finally {
+      isLoadingRef.current = false;
+    }
+  }, [page]);
+
+  const handleScroll = useCallback(
+    (event: NativeSyntheticEvent<NativeScrollEvent>) => {
+      const { layoutMeasurement, contentOffset, contentSize } =
+        event.nativeEvent;
+      const paddingToBottom = 20;
```

Comment:
Since we have to change the useRef hook, could you also put that into a constant please?